src/GovukNotify/Client/BaseClient.cs

```diff
@@ -34,55 +35,71 @@ public BaseClient(IHttpClient client, string apiKey, string baseUrl = "https://a
             this.client.BaseAddress = ValidateBaseUri(BaseUrl);
             this.client.AddContentHeader("application/json");
 
-            var productVersion = typeof(BaseClient).GetTypeInfo().Assembly.GetName().Version.ToString();
+            var productVersion = typeof(BaseClient).GetTypeInfo().Assembly.GetName().Version?.ToString();
             this.client.AddUserAgent(NOTIFY_CLIENT_NAME + productVersion);
         }
 
-        public async Task<string> GET(string url)
+        public async Task<string> GET(string url, CancellationToken cancellationToken = default)
         {
-            return await MakeRequest(url, HttpMethod.Get).ConfigureAwait(false);
+            return await MakeRequest(url, HttpMethod.Get, cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
         }
 
-        public async Task<string> POST(string url, string json)
+        public async Task<string> POST(string url, string json, CancellationToken cancellationToken = default)
         {
-            var content = new StringContent(json, Encoding.UTF8, "application/json");
-            return await MakeRequest(url, HttpMethod.Post, content).ConfigureAwait(false);
+            using var content = new StringContent(json, Encoding.UTF8, "application/json");
+            return await MakeRequest(url, HttpMethod.Post, content, cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
         }
 
-        public async Task<byte[]> GETBytes(string url)
+        public async Task<byte[]> GETBytes(string url, CancellationToken cancellationToken = default)
         {
-            return await MakeRequestBytes(url, HttpMethod.Get).ConfigureAwait(false);
+            return await MakeRequestBytes(url, HttpMethod.Get, cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
         }
 
-        public async Task<byte[]> MakeRequestBytes(string url, HttpMethod method, HttpContent content = null) {
-            var response = SendRequest(url, method, content).Result;
-
-            var responseContent = await response.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
-
+        public async Task<byte[]> MakeRequestBytes(string url, HttpMethod method, HttpContent content = null, CancellationToken cancellationToken = default) {
+            var response = await SendRequest(url, method, content, cancellationToken);
+
+#if NET6_0_OR_GREATER
+            var responseContent = await response.Content.ReadAsByteArrayAsync(cancellationToken)
+                .ConfigureAwait(false);
+#else
+            var responseContent = await response.Content.ReadAsByteArrayAsync()
+                .ConfigureAwait(false);
+#endif
```

Comment:
Passing cancellation tokens when reading content is only supported in .NET standard 2.1 and up. This project currently targets .net standard 2.0 and .net 6, so this conditional compilation should work.